### PR TITLE
dialects: Add custom printing to arith binary ops

### PR DESCRIPTION
**Pull request by PapyChacal**
Add custom syntax printing to arith's binary operations.
Honestly, I wasn't sure how to extend the tests here; `tests/filecheck/dialects/arith/arith_ops.mlir` is already parsing custom syntax, and printing custom syntax to a second xdsl-opt who prints it in generic syntax. Should we add a more explicit test than this?

### CODE REVIEW

**Comment by PapyChacal** [comment]
So I can't solve this pytest --nbval error currently, I don't get it, if someone does please tell or commit directly! Somehow the linebreaks are failing.

**Comment by math-fehr** [comment]
Hopefully this should fix it!

**Review by math-fehr — APPROVED** [review]
Nice!
Again, I think that we only need to have a single test for parsing/printing, which is:
`custom -> custom -> generic`.
This way, we test that we can roundtrip custom parsing, and then that the generic operation is what we expect the custom syntax to be.
I don't think that we need anything else in term of testing for syntax.

**Comment by PapyChacal** [comment]
Thanks for your thought on testing those!
I agree with it. Though I feel we should document it and discuss it over Zulip, before somebody disagrees after we invest in this direction? 🙂

File: tests/dialects/test_arith.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
 from xdsl.ir import Attribute
 from xdsl.utils.exceptions import VerifyException
 
-_BinOpT = TypeVar("_BinOpT", bound=BinaryOperation)
+_BinOpT = TypeVar("_BinOpT", bound=Attribute)
```
Comment (superlopuh) [inline tests/dialects/test_arith.py:52]: sus

**Comment by PapyChacal** [comment]
On a second thought, it's very possible that it was documented or extensively discussed already and that I missed it or forgot about it ? 🤔

**Comment by PapyChacal** [comment]
> There is some PR stacking going on here, right? Could you change the base of this PR, s.t. we get a clean diff to review?
Right, sorry about that, updated now

**Comment by math-fehr** [comment]
I'm not sure it's written anywhere.
I'll try to take some time today or tomorrow trying to come up with a document for it, and then from there we can edit it!

**Review by webmiche — APPROVED** [review]

**Pull request merged**